Libraries/DirectMLX.h

```diff
@@ -2660,13 +2660,63 @@ namespace dml
         return output;
     }
 
-    // 
-    // TODO: SpaceToDepth (DML_OPERATOR_SPACE_TO_DEPTH1)
-    // 
+    inline Expression SpaceToDepth(
+        Expression input,
+        uint32_t blockSize,
+        DML_DEPTH_SPACE_ORDER order = DML_DEPTH_SPACE_ORDER_DEPTH_COLUMN_ROW)
+    {
+        detail::GraphBuilder* builder = input.Impl()->GetGraphBuilder();
+        TensorDesc inputTensor = input.Impl()->GetOutputDesc();
+        dml::TensorDesc::Dimensions outputSizes = {
```

Comment:
I would `assert(inputTensor.sizes.size() == 4)` above here, because you're making unchecked accesses into it with the assumption that the size is exactly 4. (Ditto for DepthToSpace)